modules/farm_fd2/src/entrypoints/transplanting/transplanting.submitReset.e2e.cy.js

```diff
@@ -0,0 +1,227 @@
+import dayjs from 'dayjs';
+
+describe('Transplanting: Submit/Reset Buttons component', () => {
+  beforeEach(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+
+    cy.login('admin', 'admin');
+    cy.visit('fd2/transplanting/');
+    cy.waitForPage();
+  });
+
+  afterEach(() => {
+    cy.saveLocalStorage();
+    cy.saveSessionStorage();
+  });
+
+  it('Buttons exist, are visible, are enabled', () => {
+    cy.get('[data-cy="submit-button"]')
+      .should('be.visible')
+      .should('be.enabled');
+    cy.get('[data-cy="reset-button"]')
+      .should('be.visible')
+      .should('be.enabled');
+  });
+
+  function populateForm({ skipCrop = false, skipLocation = false } = {}) {
+    if (!skipCrop) {
+      cy.get('[data-cy="transplanting-picklist"]')
+        .find('[data-cy="selector-input"]')
+        .select('BROCCOLI');
+      cy.get(`[data-cy="picklist-quantity-4"]`).select(2);
+    }
+    if (!skipLocation) {
+      cy.get('[data-cy="transplanting-location"]')
+        .find('[data-cy="selector-input"]')
+        .select('ALF');
+      cy.get('[data-cy="transplanting-location"]')
+        .find('[data-cy="picker-options"]')
+        .find('input')
+        .eq(0)
+        .click();
+      cy.get('[data-cy="transplanting-location"]')
+        .find('[data-cy="picker-options"]')
+        .find('input')
+        .eq(3)
+        .click();
+    }
+
+    cy.get(
+      '[data-cy="transplanting-soil-disturbance-accordion-title"]'
+    ).click();
+    cy.get('[data-cy="equipment-selector-1"]')
+      .find('[data-cy="selector-input"]')
+      .select('Tractor');
+
+    cy.get('[data-cy="soil-disturbance-depth"]')
+      .find('[data-cy="numeric-input"]')
+      .clear();
+    cy.get('[data-cy="soil-disturbance-depth"]')
+      .find('[data-cy="numeric-input"]')
+      .type('6');
+    cy.get('[data-cy="soil-disturbance-speed"]')
+      .find('[data-cy="numeric-input"]')
+      .clear();
+    cy.get('[data-cy="soil-disturbance-speed"]')
+      .find('[data-cy="numeric-input"]')
+      .type('3');
+
+    cy.get('[data-cy="comment-input"]').clear();
+    cy.get('[data-cy="comment-input"]').type('Test comment');
+    cy.get('[data-cy="comment-input"]').blur();
+  }
+
+  it('Invalid date disables submit', () => {
+    populateForm();
+    cy.get('[data-cy="submit-button"]').should('be.enabled');
+    cy.get('[data-cy="date-input"]').clear();
+    cy.get('[data-cy="submit-button"]').click();
+    cy.get('[data-cy="submit-button"]').should('be.disabled');
```

Comment:
There is improvement to these tests that I have thought of since writing the originals in Tray Seeding and Direct seeding.  After checking that an invalid value disables the submit button, the value should be changed back to a valid value to ensure that the submit button is then re-enabled.  This addition ensures that the logic that re-enables the button works as well as the logic that disables it.

This small addition should be added to each of the tests in this file.